src/util/antlr/ANTLRErrorHandling.cpp

```diff
@@ -27,13 +27,19 @@ ExceptionMetadata generateMetadata(antlr4::ParserRuleContext* ctx) {
 }
 
 // _____________________________________________________________________________
-void ThrowingErrorListener::syntaxError(
-    [[maybe_unused]] antlr4::Recognizer* recognizer,
-    [[maybe_unused]] antlr4::Token* offendingSymbol, size_t line,
-    size_t charPositionInLine, const std::string& msg,
-    [[maybe_unused]] std::exception_ptr e) {
-  throw ParseException{
-      absl::StrCat("Token \"", offendingSymbol->getText(), "\" at line ", line,
-                   ":", charPositionInLine, " ", msg),
-      generateMetadata(recognizer, offendingSymbol, line, charPositionInLine)};
+void ThrowingErrorListener::syntaxError(antlr4::Recognizer* recognizer,
+                                        antlr4::Token* offendingSymbol,
+                                        size_t line, size_t charPositionInLine,
+                                        const std::string& msg,
+                                        [[maybe_unused]] std::exception_ptr e) {
+  if (offendingSymbol) {
+    throw ParseException{
+        absl::StrCat("Token \"", offendingSymbol->getText(), "\" at line ",
+                     line, ":", charPositionInLine, " ", msg),
+        generateMetadata(recognizer, offendingSymbol, line,
+                         charPositionInLine)};
+  } else {
+    throw ParseException{
+        absl::StrCat("line ", line, ":", charPositionInLine, " ", msg)};
```

Comment:
Is the lack of a character or sequence a good reason not to generate exception metadata also for this case?

Reply:
Somewhat. The problem here is that the start index is annoying to calculate in this case. (The line and position in the line has to be converted to a index in the whole string.) Furthermore changing this requires small further changes in other places because this would introduce the case that an `ExceptionMetadata` only has an start but not an stop index.

Reply:
1. I would suggest adding code for calculating the start index from the line etc. This will then be really convenient for visualization. Maybe we can then even at some point including the `line` in the error message, as soon as all our tools visualize this position (not here yet though).
2. I think you can just set `stopIndex == `startIndex` in this case. It would be semantically correct for lexing. Some convenience features like "If it's a lexing error and there's only a single symbol, then mark some additional characters to increase visibility" should be implemented in the UI.

Reply:
Maybe we could even make the `line` and `position in line` also part of the metadata, then tools have the choice on the way these are displayed. Are the `line` and `position in line` with respect to the same query string as the `startIndex` and `stopIndex`?

Reply:
Yes `line` and `charPositionInLine` are with respect to the same query string as the indices. `line` and `charPositionInLine` seem to be always present so there should be little harm in adding them. They are not needed at the moment. But it would not add another case where information are present/not present/partially present.

Reply:
1. The line and position in the line are already included in the textual error messages (while the indices are not).
2. This would require some mechanism to distinguish parser from lexer errors. Having a boolean flag in the metadata or another Exception for this come to my mind spontaneously.

Reply:
`line` and `charPositionInLine` are now collected but not exposed with the error JSON. They can exposed once we have a usage for them.